server/src/main/java/io/druid/segment/loading/SegmentLoaderLocalCacheManager.java

```diff
@@ -104,53 +104,71 @@ public File getSegmentFiles(DataSegment segment) throws SegmentLoadingException
   {
     StorageLocation loc = findStorageLocationIfLoaded(segment);
 
-    final File retVal;
+    File retVal = null;
 
     if (loc == null) {
-      Iterator<StorageLocation> locIter = locations.iterator();
-      loc = locIter.next();
+      Iterator<StorageLocation> locIter = Sets.newTreeSet(locations).iterator();
```

Comment:
Could you please put a comment documenting that this transformation makes locations sorted from most empty to most full? Before it was traceable from the code because `mostEmty()` method was called, now it is not clear at all.

Or, just make `locations` field a `SortedSet` in the first place?